### DISCUSSION TITLE
trivy scan failed on all branches after I removed a secret file from GitHub

### DISCUSSION THREAD

**Discussion by jianmingtu** [aquasecurity/trivy#5534] [Q&A]
### Question

My problem is that a trivy scan failed after I removed a secret file from GitHub.

In the https://github.com/bcgov/jag-crdp/ repository, I accidentally 'git push' a secret file called 'new_secret' into the branch 'feature/CEIS-1936'. I followed a tutorial https://rajeevpentyala.com/2023/08/15/step-by-step-using-bfg-repo-cleaner-tool-to-clean-up-the-git-history/ and then ran the command 'java -jar bfg.jar --delete-files new_secret jag-crdp-17'. After 'git push', the secret file 'new_secret' did not show in the branch commits' history(https://github.com/bcgov/jag-crdp/commits/feature/CEIS-1936).

C:\jag\bfg\jag-crdp-17>java -jar bfg.jar -delete-files new_secret jag-crdp-17
C:\jag\bfg\jag-crdp-17>git reflog expire --expire=now --all && git gc --prune=now --aggressive
C:\jag\bfg\jag-crdp-17>git push --force

However, a GitHub trivy scan action (https://github.com/bcgov/jag-crdp/actions/runs/6791645895) failed on trivy scan and the command 'aquasecurity/trivy-action@master' exits 1.  Trivy scan did not show why it failed. The trivy scan failed not only on my current branch but also on all other branches.

I do not understand why trivy scan fails (I think the scan should only scan the jag-crdp-transmit-receiver image and the image has no HIGH/SEVERE vulnerabilities).  

My question is: does the trivy also scan outside of the image and find the secret file in GitHub not being permanently removed from GitHub? How do I know why it fails?

Thank you!
Jimmy Tu

![p1](https://github.com/aquasecurity/trivy/assets/49298757/18361885-efbc-428d-ba5b-c24327226bbd)
![p2](https://github.com/aquasecurity/trivy/assets/49298757/1a81384b-c136-4c78-8b19-304e0e1bd6b5)

### Target

Git Repository

### Scanner

Secret

### Output Format

SARIF

### Mode

Client/Server

### Operating System

FROM eclipse-temurin:17-jre-alpine

### Version

```bash
- name: Run Trivy vulnerability scanner for jag-crdp-transmit-receiver image
        uses: aquasecurity/trivy-action@master
        with:
          scan-type: image
          image-ref: docker.io/crdp/jag-crdp-transmit-receiver:${{ github.sha }}
          format: sarif
          output: trivy-results.sarif
          exit-code: 1
          ignore-unfixed: true
          severity: HIGH,CRITICAL
```

**Comment by nikpivkin** [comment]
Hi @jianmingtu !

I scanned `jag-crdp-transmit-receiver` image and found high and critical vulnerabilities:

```bash
cat report.json| jq '.Results[].Vulnerabilities[] | select(.Severity == "CRITICAL" or .Severity == "HIGH") | .Title'
"Security Bypass With Wildcard Pattern Matching on Cloud Foundry"
"Spring Boot Welcome Page DoS Vulnerability"
"path wildcard leads to security bypass"
"Spring Expression DoS Vulnerability"
"spring: HttpInvokerServiceExporter readRemoteInvocation method untrusted java deserialization"
"Constructor Deserialization Remote Code Execution"
```

The scan results are uploaded to the [security page](https://docs.github.com/en/code-security/code-scanning/managing-code-scanning-alerts/managing-code-scanning-alerts-for-your-repository).

**Comment by nikpivkin** [comment]
@jianmingtu This doesn't address the issue, but I looked at your workflow and noticed that if there are security issues when scanning an image, subsequent images will not be scanned until the security issues are fixed.

**Comment by jianmingtu** [comment]
Thank you nikpivkin,  
I ran trivy locally but did not find HIGH,CRITICAL issues. 
how did you get high and critical vulnerabilities? And, I do not understand what is wrong with your posted vulnerability report and how to fix the issues.

here is my commands which are similar to the GitAction workflow:
/mnt/c/jag/jag-crdp-17$ mvn -ntp clean package
/mnt/c/jag/jag-crdp-17$ mvn -ntp clean verify
/mnt/c/jag/jag-crdp-17$ docker build -t docker.io/crdp/jag-crdp-transmit-receiver ./jag-crdp-transmit-receiver
/mnt/c/jag/jag-crdp-17$ trivy image docker.io/crdp/jag-crdp-transmit-receiver

![image](https://github.com/aquasecurity/trivy/assets/49298757/039320c4-f571-4ed6-bab0-4538555f4168)
